### PR TITLE
Chimera/add dependencies to tuist edit

### PR DESCRIPTION
**Pull request by laxmorek**
### Short description 📝

Working on #1674

### Solution 📦

Adds `Dependencies.swift` file to project generated using `tuist edit` command.

### Implementation 👩‍💻👨‍💻

- [x] Add loading `Dependencies.swift` in `ProjectEditorMapper`.
- [x] Update related unit tests.
- [x] Update `edit.feature` acceptance test.

### CODE REVIEW

File: Sources/TuistKit/ProjectEditor/ProjectEditorMapper.swift [hunk 1]
```diff
@@ -79,13 +81,20 @@ final class ProjectEditorMapper: ProjectEditorMapping {
                                                      targetSettings: targetSettings,
                                                      sourcePaths: [configPath])
         }
+        var dependenciesTarget: Target?
+        if let dependenciesPath = dependenciesPath {
+            dependenciesTarget = Target.editorHelperTarget(name: "Dependencies",
```
Comment (pepibumur) [inline Sources/TuistKit/ProjectEditor/ProjectEditorMapper.swift:86]: What about making `editorHelperTarget` a private method in `ProjectEditorMapper`. Considering that method is scoped to this file, I think it's more appropiate to define it here.

File: Sources/TuistLoader/Utils/ManifestFilesLocator.swift [hunk 2]
```diff
@@ -51,6 +55,11 @@ public final class ManifestFilesLocator: ManifestFilesLocating {
         return traverseAndLocate(at: at, appending: subPath)
     }
 
+    public func locateDependencies(at: AbsolutePath) -> AbsolutePath? {
+        let subPath = RelativePath("\(Constants.tuistDirectoryName)/\(Manifest.dependencies.fileName(at))")
```
Comment (pepibumur) [inline Sources/TuistLoader/Utils/ManifestFilesLocator.swift:59]: This confuses me a bit. Does `Manifest.dependencies.fileName(at)` return the name of the manifest file? I think you need to read the value from [this attribute](https://github.com/tuist/tuist/blob/master/Sources/TuistLoader/Models/Manifest.swift#L28)

**Review by pepibumur — APPROVED** [review]
Besides the small comments, your PR looks great :)

File: Sources/TuistKit/ProjectEditor/ProjectEditor.swift [hunk 3]
```diff
@@ -93,6 +93,7 @@ final class ProjectEditor: ProjectEditing {
         let projectDesciptionPath = try resourceLocator.projectDescription()
         let manifests = manifestFilesLocator.locateAllProjectManifests(at: at)
         let configPath = manifestFilesLocator.locateConfig(at: at)
+        let dependenciesPath = manifestFilesLocator.locateDependencies(at: at)
```
Comment (facumenzella) [inline Sources/TuistKit/ProjectEditor/ProjectEditor.swift:96]: Not related ot this PR, but maybe we can add a parameter name and an argument label so this reads like:
```
func edit(at editingPath: AbsolutePath, in dstDirectory: AbsolutePath) throws -> AbsolutePath {
    ...
    let dependenciesPath = manifestFilesLocator.locateDependencies(at: edititingPath)

}
```

**Review by facumenzella — APPROVED** [review]
🚀

File: Sources/TuistLoader/Utils/ManifestFilesLocator.swift (hunk 2 above)
Comment (pepibumur) [reply Sources/TuistLoader/Utils/ManifestFilesLocator.swift:59]: But `fileName(at)` is a selector, isn't it? Should we add a test for this function to make sure we are getting the expected value?

**Comment by pepibumur** [comment]
@all-contributors add @laxmorek for contributions

**Comment by pepibumur** [comment]
@all-contributors add @laxmorek  for code

File: Sources/TuistLoader/Utils/ManifestFilesLocator.swift (hunk 2 above)
Comment (laxmorek) [reply Sources/TuistLoader/Utils/ManifestFilesLocator.swift:59]: I added some tests for this method in `ManifestFilesLocatorTests` in this PR. All pass.

Comment (laxmorek) [reply Sources/TuistLoader/Utils/ManifestFilesLocator.swift:59]: you can check them [here](https://github.com/tuist/tuist/blob/a61725fa8143fc8df643aa9b4f387d60e431e319/Tests/TuistLoaderTests/Utils/ManifestFileLocatorTests.swift#L158)

Comment (laxmorek) [reply Sources/TuistLoader/Utils/ManifestFilesLocator.swift:59]: you can check them [here](https://github.com/tuist/tuist/blob/a61725fa8143fc8df643aa9b4f387d60e431e319/Tests/TuistLoaderTests/Utils/ManifestFileLocatorTests.swift#L158)

**Comment by laxmorek** [comment]
@pepibumur @facumenzella I have applied to your suggestions. Please check them :)

**Pull request merged**